src/rust-crypto/RoomEncryptor.ts

```diff
@@ -103,7 +109,21 @@ export class RoomEncryptor {
         this.prefixedLogger.debug("Sessions for users are ready; now sharing room key");
 
         const rustEncryptionSettings = new EncryptionSettings();
-        /* FIXME historyVisibility, rotation, etc */
+        /* FIXME historyVisibility, etc */
+
+        // We only support megolm
+        rustEncryptionSettings.algorithm = EncryptionAlgorithm.MegolmV1AesSha2;
```

Comment:
Two solutions here:

- Set directly the Megolm algorithm since we currently only supporting megolm and we are checking at the beginning at the function that we are using Megolm.
- Use the value from  `this.encryptionSettings`, it means that we need to convert the string in the field `this.encryptionSettings.algoritm` to the rust `EncryptionAlgorithm` enum

The second solution appears to me useless since we are only using Megolm